spec/takeaway_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'takeaway'
+
+describe Takeaway do
+  subject(:takeaway) { described_class.new }
+  subject(:order) { described_class.new }
+  let(:basket) { double :basket }
+
+  describe '#open_menu' do
+    it 'opens a list of dishes with prices' do
+      expect(takeaway.open_menu).not_to be_empty
+    end
+  end
+
+  describe '#order' do
+    it 'reports items being added to basket' do
+      expect{takeaway.order('kimchi', 2)}.to change{order.basket.length}.by(1)
+    end
+  end
+
+  describe '#order_summary' do
+    it 'creates an order summary' do
+      takeaway.order('kimchi', 2)
+      takeaway.order('salmon maki', 3)
+      expect(takeaway.order_summary).to eq("kimchi x2 = £6.00, salmon maki x3 = £16.50")
```

Comment:
you test the return value (which is meaningless in this case as it returns a string that will never be seen), but does not test the action - which is that the variable @all should be modified. If you had tested that, I suspect you would have spotted an error you had made in your  code. This would have been picked up if you had written your test first.